packages/simulation-sdk-wagmi/src/hooks/useSimulationState.ts

```diff
@@ -249,9 +262,45 @@ export function useSimulationState<
     },
   });
 
+  const vaultV2s = useVaultV2s({
+    ...parameters,
+    blockNumber: block?.number,
+    query: {
+      ...parameters.query,
+      enabled: block != null && parameters.query?.enabled,
+    },
+  });
+
+  const vaultV2Adapters = useVaultV2Adapters({
+    ...parameters,
+    blockNumber: block?.number,
+    query: {
+      ...parameters.query,
+      enabled: block != null && parameters.query?.enabled,
+    },
+  });
+
   const data = useMemo(() => {
     if (block == null) return;
 
+    for (const token of values(tokens.data)) {
+      if (!token) continue;
```

Comment:
Can we check `token == null`? I know most of the time it's ok but the fact that js also considers empty string as false is scary to me so if we only need to check it's not null I feel safer relying on pure checks